lib/web/oidcidp.go

```diff
@@ -89,6 +93,61 @@ func (h *Handler) jwksOIDC(_ http.ResponseWriter, r *http.Request, _ httprouter.
 	return &resp, nil
 }
 
+// thumbprint returns the thumbprint as required by AWS when adding an OIDC Identity Provider.
+// This is documented here:
+// https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc_verify-thumbprint.html
+func (h *Handler) thumbprint(_ http.ResponseWriter, r *http.Request, _ httprouter.Params) (interface{}, error) {
+	// Dial requires the following address format: host:port
+	cfgPublicAddress := h.cfg.PublicProxyAddr
```

Comment:
I'm guessing we are intentionally connecting to the proxy public addr instead of returning the thumbprint of the locally served cert in case TLS is being terminated at a load balancer? If so can we include that in a comment?

Reply:
This returns the thumbprint of the CA that signed the public address certificate. And not the thumbprint of the served certificate by the public address (whether that's by a load balancer or Teleport Proxy itself)

I added the following comment:
```
// Returns the thumbprint of the top intermediate CA that signed the TLS cert used to serve HTTPS requests.
// In case of a self signed certificate, then it returns the thumbprint of the TLS cert itself.

```